src/vm/wren_core.c

```diff
@@ -942,7 +942,7 @@ DEF_PRIMITIVE(num_fromString)
   double number = strtod(string->value, &end);
 
   // Skip past any trailing whitespace.
-  while (*end != '\0' && isspace(*end)) end++;
+  while (*((unsigned char*)end) != '\0' && isspace(*((unsigned char*)end))) end++;
```

Comment:
The first check can't produce the warning you're fixing, only the call to `isspace` would if it is implemented as a macro that does an array index, which is presumably the case in your headers.

Also, the cast could be done after dereferencing the pointer which looks a little cleaner:

```
while (*end != '\0' && isspace((unsigned char) *end)) end++;
```